packages/server/src/listener.js

```diff
@@ -72,4 +77,36 @@ export default class Listener {
     // Set this.listening to true
     this.listening = true
   }
+
+  async serverListen() {
+    // Prepare listenArgs
+    const listenArgs = this.socket ? { path: this.socket } : { host: this.host, port: this.port }
+    listenArgs.exclusive = false
+    // Call server.listen
+    this.server = await new Promise((resolve, reject) => {
+      const s = this._server.listen(listenArgs, error => error ? reject(error) : resolve(s))
+    })
+  }
+
+  serverErrorHandler(error) {
+    // Detect if port is not available
+    const addressInUse = error.code === 'EADDRINUSE'
+
+    // Use better error message
+    if (addressInUse) {
+      error.message = `Address \`${this.host}:${this.port}\` is already in use.`
+    }
+
+    // Listen to a random port on dev as a fallback
+    if (addressInUse && this.dev && this.port !== '0') {
+      consola.warn(error.message)
+      consola.info('Trying a random port...')
+      this._server.close()
+      this.port = '0'
+      return this.serverListen()
+    }
+
+    // Throw error
+    throw error
```

Comment:
Tip: This error is handled and should be handled by consola. The reason why not using `consola` here is that programatic usage needs to capture the listen error instead of silently skipping